core/src/main/java/com/linecorp/armeria/client/encoding/HttpDecodedResponse.java

```diff
@@ -80,10 +85,16 @@ protected HttpObject filter(HttpObject obj) {
             if (contentEncoding != null) {
                 final StreamDecoderFactory decoderFactory =
                         availableDecoders.get(Ascii.toLowerCase(contentEncoding));
-                // If the server returned an encoding we don't support (shouldn't happen since we set
-                // Accept-Encoding), decoding will be skipped which is ok.
                 if (decoderFactory != null) {
                     responseDecoder = decoderFactory.newDecoder(alloc);
+                } else {
+                    // The server returned an encoding we don't support.
+                    // This shouldn't happen normally since we set Accept-Encoding.
+                    if (strictContentEncoding) {
+                        delegate.abort(new UnsupportedEncodingException("encoding: " + contentEncoding));
```

Comment:
After aborting the `delegate`, this returns the headers and `delegate.onNext(headers)` is called.
If `delegate.onError(...)` is called before that, that's no problem. However, depending on the implementation of the delegate `StreamMessage`, I think there's a chance that the `onError` is called first.
So I think we need to introduce a new flag (or throwing an exception). The flag is checked in `onNext`(perhaps in `onSubscribe` as well?) after `filter(o);` is called.

Reply:
It might be slightly better to throw an exception.